### PR TITLE
Combined Horseback Archery Rewards

### PR DESCRIPTION
**Pull request by Roman971**
This renames the "Skip First Dampe Race" setting to "Skip First Minigame Phases", and changes it to also allow getting both rewards from the Gerudo Horseback Archery minigame at once during a single attempt if you score 1,500 points. While the implementation is different and a bit more complex, this simply applies the same idea as the Dampe Race improvement to the Horseback Archery.
Enabling the setting also changes the first text when starting the HBA minigame to make it clear that it is possible to get rewards if you score both 1,000 and 1,500 points.

To make it clear, the setting's impact on the Dampe Race behavior isn't changed at all, and this only adds the HBA rewards change to the already existing setting.

Note: I did my best on the wording of the setting and tooltip, but I'm not sure it's ideal, so feel free to suggest or make any changes in that regard.

### CODE REVIEW

File: Patches.py [hunk 1]
```diff
@@ -1256,6 +1253,12 @@ def set_entrance_updates(entrances):
         symbol = rom.sym('JABU_ELEVATOR_ENABLE')
         rom.write_byte(symbol, 0x01)
 
+    if world.no_first_minigame_phases:
```
Comment (Dwood15) [inline Patches.py]: In new code, can we try to access settings via `world.settings` instead?

**Comment by Roman971** [comment]
I haven't confirmed it yet but these changes may cause crashes on VC only, so don't merge this for the time being.

**Comment by Roman971** [comment]
The VC issue I mentioned is now fixed, and there shouldn't be any problems left in these changes.

**Comment by ThomasJRyan** [comment]
I take it this doesn't affect the bombchu bowling? If not then the "Skip First Minigame Phases" is misleading

**Comment by Roman971** [comment]
Yeah this doesn't affect bombchu bowling in any way.

As I said I'm not sure the name is ideal but I can't think of anything better.

**Comment by Roman971** [comment]
Haven't been able to look into it yet but this may be broken on N64 now, so once again don't merge this for the time being.

**Comment by Zannick** [comment]
> don't merge this for the time being.

Acknowledged. I'll come back to review when you're ready.

I might suggest "Combine Some Minigame Rounds" as a name replacement. I think it's more accurate to say the rounds or phases are combined rather than skipped because we aren't removing the rewards. (Plus if we don't have to say "First" we can more easily add "Some" to indicate it's not all minigames.)

**Comment by Roman971** [comment]
I will need confirmation from someone testing on N64 but I believe all issues should be fixed now.

**Comment by Roman971** [comment]
Can confirm this now works properly on N64, based on the feedback I got from people testing it.

**Comment by Darkunov** [comment]
Some more potential names/inspiration :

- "Combine Both Dampe and HBA Rewards"
- "Enable Efficient Minigame Clears"
- "Stack both Minigame Difficulties"
- "Allow Two Prizes on One Good Clear"
- "Skilled Clears Yield Two Prizes"

My personal favorite would be Stack Both Minigame Difficulties (or "Stack Both Minigame Difficulties Rewards", but this is getting pretty long) since I believe Dampe and HBA are the only two minigames whose difficulty escalates, unlike Bombchu Bowling.

**Comment by Zannick** [comment]
Although "Efficient Minigames" is appealing...

**Pull request merged**